nixos/modules/services/networking/pihole-ftl.nix

```diff
@@ -0,0 +1,330 @@
+{ pkgs
+, lib
+, config
+, ...
+}:
+
+with lib;
+
+let
+  cfg = config.services.pihole-ftl;
+  confFile = "${cfg.stateDirectory}/pihole-FTL.conf";
+  logFile = "/var/log/pihole/FTL.log";
+  debugFile = "/var/log/pihole/pihole.log";
+
+  ftlConf = (pkgs.formats.keyValue { }).generate "pihole-FTL.conf" ({
+    PRIVACYLEVEL = cfg.privacyLevel;
+  } // cfg.extraConfig);
+
+  # Keys in setupVars.conf, reverse-engineered from the installation script,
+  # adapted to use the dnsmasq module settings.
+  setupVars =
+    let
+      dnsmasqConfig = config.services.dnsmasq;
+      uiConfig = config.services.pihole-adminlte;
+      settings = dnsmasqConfig.settings;
+      boolSetting = attribute: if hasAttr attribute settings then (elemAt (getAttr attribute settings) 0) else false;
+      dhcpEnabled = hasAttr "dhcp-range" settings;
+    in
+    {
+      DHCP_ACTIVE = false;
+      DNSMASQ_LISTENING =
+        if boolSetting "local-service" then "local"
+        else if boolSetting "bind-interfaces" then "bind"
+        else "all";
+      DNSSEC = boolSetting "dnssec";
+      DNS_BOGUS_PRIV = boolSetting "bogus-priv";
+      DNS_FQDN_REQUIRED = boolSetting "domain-needed";
+      QUERY_LOGGING = boolSetting "log-queries";
+      TEMPERATUREUNIT = uiConfig.temperatureUnit;
+      WEBTHEME = uiConfig.theme;
```

Comment:
These two settings come from the pihole-admin module. It's a bit dodgy that this single config file is shared between the two.

I think patching pihole-admin to use a separate config file will be difficult, since it accesses both its own and pihole-ftl's settings here.